### PR TITLE
[Perf] Add a CPU-based training workload

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

Our scalability test requires some CPU-based workloads. This PR uses CPU to train a MNIST model. I also can run the RayJob successfully on my devbox.

Screenshot 2024-05-03 at 10 48 05 AM

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    # [4] Start distributed training
+    # Run `train_func_per_worker` on all workers
+    # =============================================
+    result = trainer.fit()
+    print(f"Training result: {result}")
+
+
+if __name__ == "__main__":
+    train_fashion_mnist(num_workers=4)
```
Comment (andrewsykim) [inline ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py:163]: Thoughts on making num_workers configurable with an environment variable? That would it make it easier to tweak in the perf tests

Comment (andrewsykim) [reply ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py:163]: Same with CPUs for resources_per_worker

**Comment by andrewsykim** [comment]
LGTM! Tested this locally and confirmed it works. Thanks @kevin85421!

File: ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py (hunk 1 above)
Comment (kevin85421) [reply ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py:163]: Good idea

Comment (kevin85421) [reply ray-operator/config/samples/pytorch-mnist/ray_train_pytorch_mnist.py:163]: I will ask others to work on it.

**Review by jjyao — APPROVED** [review]

**Pull request merged**